pkg/invariantlibrary/historicaldata/alert_types.go

```diff
@@ -79,9 +80,15 @@ func NewAlertMatcherWithHistoricalData(data map[AlertDataKey]AlertStatisticalDat
 
 func (b *AlertBestMatcher) bestMatch(key AlertDataKey) (AlertStatisticalData, string, error) {
 	exactMatchKey := key
+	logrus.WithField("alertName", key.AlertName).Infof("searching for bestMatch for %+v", key.JobType)
+	logrus.Infof("historicalData has %d entries", len(b.HistoricalData))
 
-	if percentiles, ok := b.HistoricalData[exactMatchKey]; ok && percentiles.JobRuns >= minJobRuns {
-		return percentiles, "", nil
+	if percentiles, ok := b.HistoricalData[exactMatchKey]; ok {
+		if percentiles.JobRuns > minJobRuns {
+			logrus.Infof("found exact match: %+v", percentiles)
+			return percentiles, "", nil
+		}
+		PreviousReleaseUpgrade(key.JobType)
```

Comment:
Reviewed with Deep, we aren't actually doing anything with the updated matcher here.  Need to consolidate the logic in the loop below for reusability and check it check for results here as well.  Same for disruption.